### PR TITLE
Add REST API for role validation

### PR DESCRIPTION
**Pull request by vithu30**
This PR contains changes to add Publisher REST APIs to 

- check whether if a given role exists in the system
- check whether the logged-in user has the given role

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/gen/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/MeApi.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+@Api(description = "the me API")
+@Consumes({ "application/json" })
+@Produces({ "application/json" })
+
+
+public class MeApi  {
+
+  @Context MessageContext securityContext;
+
+MeApiService delegate = new MeApiServiceImpl();
```
Comment (AmaliMatharaarachchi) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/gen/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/MeApi.java]: Formatting issue. Shall we add a tab space before starts the line

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/MeApiServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+import org.wso2.carbon.apimgt.rest.api.publisher.v1.MeApiService;
+import org.apache.cxf.jaxrs.ext.MessageContext;
+import org.wso2.carbon.apimgt.rest.api.util.utils.RestApiUtil;
+import org.wso2.carbon.user.api.UserStoreException;
+import javax.ws.rs.core.Response;
+
+public class MeApiServiceImpl implements MeApiService {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/MeApiServiceImpl.java]: Let's add class comments and method comments

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/RolesApiServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+import org.wso2.carbon.apimgt.rest.api.publisher.v1.RolesApiService;
+import org.apache.cxf.jaxrs.ext.MessageContext;
+import org.wso2.carbon.apimgt.rest.api.util.utils.RestApiUtil;
+import javax.ws.rs.core.Response;
+
+public class RolesApiServiceImpl implements RolesApiService {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/RolesApiServiceImpl.java]: Class comment and method comments

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/gen/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/MeApi.java (hunk 1 above)
Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/gen/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/MeApi.java]: This file contains auto-generated code..

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/MeApiServiceImpl.java (hunk 2 above)
Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/MeApiServiceImpl.java]: Added comments in 5f19453

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/RolesApiServiceImpl.java (hunk 3 above)
Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/RolesApiServiceImpl.java]: Added comments in 5f19453

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 4]
```diff
@@ -5939,6 +5996,16 @@ parameters:
     required: true
     type: string
 
+# Role ID
+  roleId:
+    name: roleId
+    in: path
+    description: |
+      The Base 64 URL encoded role name. If the role existed in secondary user-store, role name should be specified as
+      {user-store-name}/{role-name}.
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Shall we correct the wording as:
```
The Base 64 URL encoded role name. If the role existed in secondary user-store, role name should be specified as Base 64 URL encoded( {user-store-name}/{role-name} ).
```

Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Updated the wordings

**Pull request merged**